### PR TITLE
explicit declare unsupported operations

### LINKED ISSUE #677: Explicit declare the unsupported data type
See #645 , we need to explicit declare the un-supported data type.

### PR DESCRIPTION
**Pull request by adrian-wang**
## What changes were proposed in this pull request?

Explicit declare unsupported operations.
this fixes #677 

## How was this patch tested?

message change

### CODE REVIEW

**Comment by linhong-intel** [comment]
LGTM. Thanks @adrian-wang

File: src/main/java/org/apache/parquet/hadoop/VectorizedOapRecordReader.java [hunk 1]
```diff
@@ -311,7 +311,7 @@ protected void initializeInternal() throws IOException, UnsupportedOperationExce
       for (int i = 0; i < requestedSchema.getFieldCount(); ++i) {
         Type t = requestedSchema.getFields().get(i);
         if (!t.isPrimitive() || t.isRepetition(Type.Repetition.REPEATED)) {
-          throw new UnsupportedOperationException("Complex types not supported.");
+          throw new UnsupportedOperationException("Complex types" + t.getName() + "not supported.");
```
Comment (gczsjdy) [inline src/main/java/org/apache/parquet/hadoop/VectorizedOapRecordReader.java]: :nit More spaces?

File: src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java [hunk 2]
```diff
@@ -235,12 +235,12 @@ public ArrayData getArray(int ordinal) {
 
     @Override
     public MapData getMap(int ordinal) {
-      throw new UnsupportedOperationException();
+      throw new UnsupportedOperationException("get map");
```
Comment (gczsjdy) [inline src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java]: :nit `get map not supported`?

Comment (adrian-wang) [reply src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java]: the message looks like UnsupportedOperationException: get map. I suppose that is enough.

**Pull request merged**